Sofa/framework/Helper/src/sofa/helper/Utils.cpp

```diff
@@ -257,34 +243,76 @@ const std::string& Utils::getUserLocalDirectory()
             // "Failed to get settings folder"
             return std::string("");
         }
-        
+
+#else // Linux: ${HOME}
+
+    const char* homeDir;
+
+    // if HOME is defined
+    if ((homeDir = std::getenv("HOME")) == nullptr)
+    {
+        // else system calls are used
+        homeDir = getpwuid(getuid())->pw_dir;
+    }
+
+    return std::string(homeDir);
+
+#endif
+    };
+
+    static std::string homeDir = FileSystem::cleanPath(computeUserHomeDirectory());
+    return homeDir;
+}
+
+const std::string& Utils::getSofaDataDirectory()
+{
+    constexpr std::string_view sofaDataDirSuffix = "SOFAData";
+
+    static std::string sofaDataDirectory = FileSystem::cleanPath(FileSystem::findOrCreateAValidPath(
```

Comment:
Maybe the error on the linux unit-tests come from the deprecated usage of `findOrCreateAValidPath`?
In any case `findOrCreateAValidPath()` is deprecated and will be removed so it needs to be replaced.

```suggestion
    static std::string sofaDataDirectory = FileSystem::cleanPath(FileSystem::ensureFolderExists(
```